src/common/interfaces/IGateway.sol

```diff
@@ -221,10 +221,10 @@ interface IGateway is IMessageHandler, IMessageSender, IGatewayHandler {
     /// @notice Used to calculate overall cost for bridging a payload on the first adapter and settling
     ///         on the destination chain and bridging its payload proofs on n-1 adapter
     ///         and settling on the destination chain.
-    /// @param  payload Used in gas cost calculations.
+    /// @param  actions Used in gas cost calculations.
     /// @dev    Currenly the payload is not taken into consideration.
     /// @return total Total cost for sending one message and corresponding proofs on through all adapters
-    function estimate(uint16 centrifugeId, bytes calldata payload) external view returns (uint256 total);
+    function estimate(uint16 centrifugeId, bytes4[] memory actions) external view returns (uint256 total);
```

Comment:
I think sadly bytes4 is not enough. For example, for `Hub.updateContract.selector` I do not know the message size/cost associated to it because it's dynamic. 

By now, we can assume the worst case, but maybe we can also parameterize and use some tuple `(bytes4, kind)` to specify some special behavior in the action.

Reply:
I think is not horrible to assume the worst case, because the unused gas is refunded